lib/utilities/app.state.dart

```diff
@@ -25,170 +29,200 @@ class AppStateHandler extends ChangeNotifier {
+    } else {
+      setLoggedOutState();
+    }
   }
 
-  /// App is authenticated | User is locked
-  /// Hive is closed
-  void setLoggedOutState(BuildContext context) {
+  /// App is authenticated | User is logged out
+  void setLoggedOutState() {
     if (currentState == AppState.loggedOut) return;
+    syncTimer?.cancel();
     // clear AppLoggin | clear hive
     logout();
     currentState = AppState.loggedOut;
   }
 
+  /// Returns true when relogin was successful
+  ///
+  /// See [AppState.relogin] for more information
+  Future<bool> setReloginState() async {
+    final thirtyDaysAgo = DateTime.now().subtract(const Duration(days: 30));
+    final tooLongOffline = getLastLoginCheck().isBefore(thirtyDaysAgo) ||
+        getLastNamiSync().isBefore(thirtyDaysAgo);
+    var showLogin = true;
+    if (tooLongOffline) {
+      showSnackBar(navigatorKey.currentContext!,
+          "Sie waren zu lange offline. Bitte melden Sie sich erneut an.");
+    } else {
+      showLogin = await showConfirmationDialog(
+        "Sitzung abgelaufen",
+        "Ihre Sitzung ist abgelaufen. Bitte melden Sie sich erneut an.",
+      );
+    }
+    if (showLogin) {
+      currentState = AppState.relogin;
+
+      final reloginSuccessful = await navigatorKey.currentState!.push<bool?>(
+        MaterialPageRoute(builder: (context) => const LoginScreen()),
+      );
+      setReadyState();
+      return reloginSuccessful ?? false;
+    } else {
+      return false;
+    }
+  }
+
   /// App is authenticated | User is logged in
-  /// Hive is open
-  Future<void> setLoadDataState(BuildContext context,
-      {bool loadAll = false}) async {
-    if (currentState == AppState.loadData) return;
-    ValueNotifier<bool?> loginProgressNotifier = ValueNotifier(null);
+  Future<void> setLoadDataState({
+    bool loadAll = false,
+    background = false,
+  }) async {
     ValueNotifier<List<AllowedFeatures>> rechteProgressNotifier =
         ValueNotifier([]);
     ValueNotifier<String?> gruppierungProgressNotifier = ValueNotifier(null);
     ValueNotifier<bool?> metaProgressNotifier = ValueNotifier(null);
     ValueNotifier<bool?> memberOverviewProgressNotifier = ValueNotifier(null);
     ValueNotifier<double> memberAllProgressNotifier = ValueNotifier(0.0);
-    ValueNotifier<bool> statusGreenNotifier = ValueNotifier(true);
-
-    currentState = AppState.loadData;
-
-    Navigator.push(context, MaterialPageRoute(builder: (context) {
-      return LoadingInfoScreen(
-        loginProgressNotifier: loginProgressNotifier,
-        rechteProgressNotifier: rechteProgressNotifier,
-        gruppierungProgressNotifier: gruppierungProgressNotifier,
-        metaProgressNotifier: metaProgressNotifier,
-        memberOverviewProgressNotifier: memberOverviewProgressNotifier,
-        memberAllProgressNotifier: memberAllProgressNotifier,
-        statusGreenNotifier: statusGreenNotifier,
-        loadAll: loadAll,
+
+    if (background) {
+      syncState = SyncState.loading;
```

Comment:
Wenn Background, dann bitte mit Toast kurze Info über Start Sync und ende Sync

Reply:
Das meinte ich mit TODO für eine weitere PR. Ich hatte eher an ein Icon in der AppBar gedacht, aber eine Snackbar ist auch machbar.

Reply:
Der Sync könnte ja überlapp passieren, da hast du ggf gar keine AppBar

Reply:
Eigentlich hat ja jede Seite ein Scaffold und dann auch eine AppBar. Aber eine Snackbar ist auch simpler zu implementieren :+1: